compysition/actor.py

```diff
@@ -215,34 +248,38 @@ def _send(self, queue, event):
         queue.put(event)
         sleep(0)
 
-    def __consumer(self, function, queue):
+    def __consumer(self, function, queue, timeout=10, ensure_empty=True):
         '''Greenthread which applies <function> to each element from <queue>
         '''
 
         self.__run.wait()
 
         while self.loop():
             queue.wait_until_content()
-            try:
-                event = queue.get(timeout=10)
-            except QueueEmpty:
-                pass
-            else:
-                if self.__blocking_consume:
-                    self.__do_consume(function, event, queue)
-                else:
-                    self.threads.spawn(self.__do_consume, function, event, queue, restart=False)
+            self.__process_consumer_event(function=function, queue=queue, timeout=timeout)
 
-        while True:
-            if queue.qsize() > 0:
-                try:
-                    event = queue.get()
-                except QueueEmpty as err:
-                    break
-                else:
-                    self.threads.spawn(self.__do_consume, function, event, queue, restart=False)
+        try:
+            while ensure_empty and queue.qsize() > 0:
```

Comment:
Do you know if this second `while` loop is necessary? I had this block commented out for a while and never hit any issues, and don't know of a way that we get to the second block.

Reply:
Disregard this one, can see we do hit it in some cases